# Rubric

Instructions: DELETE this cell before you submit via a `git push` to your repo before deadline. This cell is for your reference only and is not needed in your report. 

Scoring: Out of 18 points in total.  15 points in the report.  3 points in the video summary of your report for a non-technical audience.

For the report section Title & Abstract trhough Data Wrangling:
- Each Developing  => -0.5 pts
- Each Unsatisfactory/Missing => -1 pts

For the report section Data Visualizations through Documentation and Written Communication:
- Each Developing  => -1 pts
- Each Unsatisfactory/Missing => -2 pts

For the video:
- Developing  => -1.5 pts
- Unsatisfactory/Missing => -3 pts

Until the score is 0

|                                                                                    | Unsatisfactory                                                                                                                                                                                                                                                                                      | Developing                                                                                                                                                                                                                                                                                                                          | Proficient                                                                                                                                                                                                                                                    | Exellent                                                                                                                                                                                                                                                                                                                                                          |
| ---------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Title & abstract                                                                   | Title or abstract is confusing or fails to offer important details about the question, variables, context, or methods of the project.                                                                                                                                                               | Title or abstract lacks relevance or fails to offer appropriate details about the question, variables, context, or methods of the project.                                                                                                                                                                                          | Title and abstract are relevant, offering sufficient details about the research project.                                                                                                                                                                      | Title and abstract are informative, succinct, and clear. It offers specific details about the question, variables, context, and proposed methods of the study.                                                                                                                                                                                                    |
| Background                                                                         | Did not have at least 3 reliable and relevant sources. Or relevant sources were not used in relevant ways                                                                                                                                                                                           | A key component was not connected to the research literature. Selected literature was from unreliable sources. Literary supports were vague or ambiguous.                                                                                                                                                                           | Key research components were connected to relevant, reliable literature.                                                                                                                                                                                      | Narrative integrates critical and logical details from the peer-reviewed or other professional research literature. Each key research component is grounded to the literature. Attention is given to different perspectives, threats to validity, and opinion vs. evidence.                                                                                       |
| Research question & hypothesis                                                     | Research issue remains unclear. The research purpose, questions, hypotheses, definitions or variables and controls are still largely undefined, or when they are are poorly formed, ambiguous, or not logically connected to the description of the problem. Unclear connections to the literature. | Research issue is identified, but statement is too broad or fails to establish the importance of the problem. The research purpose, questions, hypotheses, definitions or variables and controls are poorly formed, ambiguous, or not logically connected to the description of the problem. Unclear connections to the literature. | Identifies a relevant research issue. Research questions are succinctly stated, connected to the research issue, and supported by the literature. Variables and controls have been identified and described. Connections are established with the literature. | Presents a significant research problem related to the chemical sciences. Articulates clear, reasonable research questions given the purpose, design, and methods of the project. All variables and controls have been appropriately defined. Proposals are clearly supported from the research and theoretical literature. All elements are mutually supportive. |
| Data description                                                                   | Did not have references to relevant data sources for this problem. Did not describe the data obtained at those sources                                                                                                                                                                              | A key data source was not referenced or described in satisfactory level of detail                                                                                                                                                                                                                                                   | All relevant data sources were referenced and described in terms of their key variables and size                                                                                                                                                              | Multiple data sources for each aspect of the project, All data sources are fully described and referenced. The details of the descriptions also make it clear how they support the needs ot the project.                                                                                                                                                          |
| Data wrangling                                                                     | Data obviously not tidy and not clean. Or even if they are clean/tidy, there was no demonstration of that fact                                                                                                                                                                                      | Data has some minor problems with tidiness or cleanliness. Potentially these are not problems but it wasn't fully demonstrated to be the case in the code                                                                                                                                                                           | Data are made tidy and clean, or are clearly demonstrated to be so from download                                                                                                                                                                              | Data wrangling is complete, but also particularly thoroughly done for a very tough case. Often free text parsing is a bear like this, and doing it well can mean a million edge cases.                                                                                                                                                                            |
| Ethics                                                                             | No effort or just says we have no ethical concerns                                                                                                                                                                                                                                                  | Minimal ethical section; probably just talks about data privacy and no unintended consequences discussion. Ethical concerns raised seem irrelevant.                                                                                                                                                 | The ethical concerns described are appropriate and sufficiently                                                                                                                                                                                                                                                                     | Ethical concerns are described clearly and succinctly. This was clearly a thorough and nuanced approach to the issues                                                                                                                                                                                                          |
| Data visualizations                                                                | No plots. Or most plots don't capture useful information, or their display of information is difficult to understand. Legends, labels, axes, titles, captions, etc of most plots may be poor or missing. Most figures are not referenced in the main text.                                          | Fewer than 3 relevant plots (irrelevant to the question plots do not count).  Or a few of the plots don't make sense or capture useful information. Legends, labels, axes, titles, captions, etc are illegible or inadequately describe the content in a few figures. Or a few figures are not referenced or explained inside the main text. | Plots almost all make sense and capture useful information. Legends, labels, axes, titles, captions, etc are clearly legible and adequately describe the content of the figure. Figures are referenced or explained inside the main text and not just left hanging.      | Plots are beautiful and information dense. Legends, labels, axes, titles, captions, etc clearly describe the content of the figure. Figures are well integrated with the text.                                                                                                                                                                                    |
| Data analysis and results                                                          | No analyses beyond EDA style explorations of the dataset. Or analyses presented do not help address the question/hypothesis. Or most of the analyses presented are not technically approriate (eg. using parametric methods for non-parametric data).                                              | Only one analysis beyond EDA. Or a few analyses presented do not address the question/hypothesis. Or a few of the analyses presented are not technically approriate                                                                                                                                                                  |  At least two analyses beyond EDA. That is, the project uses two different techniques (e.g., inference vs machine learning; or t-test vs regression; or entire dataset vs stratified) or two different proxies for the same concept are investigated.  Each analysis addressed the question/hypothesis and is technically approriate. | Many different analyses are used. Each analysis addressed an aspect of the question/hypothesis and is technically approriate. Perhaps they use multiple datasets, or do something else to go the extra mile and address the reliability of the results                                            |
| Conclusion and discussion                                                          | Doesn't have a narrative/conclusions of what the report accomplished. Or very minimal conclusions, like "X was best" without anything else                                                                                                                                                          | Has narrative/conclusions, but it feels perfunctory and lacks context. Especially if the limitations and ethics sections are largely junk or missing                                                                                                                                                                                | Properly recapitulates the results and provides context for them. Has a sense of the limitatins and ethical issues.                                                                                                                                           | Clearly recapitulates the results and provides context, including the literature/background Has a sense of what to do next, and has good explorations of the limitations and ethics.                                                                                                                                                                              |
| Documentation and written communication                                            | Code documentation is non-existent or perfunctory. Writing has usage errors, poor fluency, and poor rhetoric. If doc/writing are both like this you get unsatisfacory. If one is unsatisfactory and the other is developing you still get unsatisfactory overall.                                   | Code documentation is only partially there. Writing has some usage errors, some poor fluency, or some poor rhetoric. If doc/writing are both like this you get developing. If one is developing and the other is satisfactory you still get developing.                                                                             | Code is documented throughout. Writing has good fluency, good rhetoric, and at most a few minor errors.                                                                                                                                                       | Code documentation is highly informative with docstrings. Writing is high quality. If both doc/writing are like this you get excellent. If one is satisfactory and the other excellent you still get satisfactory.                                                                                                                                                |
| Video (developing -1.5 points, unsatisfactory -3 points)                           | Video well outside the 3-5min time range. Does not communicate major results at a non-technical level. Has major technical issues such as no sound or not available at link. Unsatisfactory if 2/3 of these are true                                                                                | Video just outside the 3-5min time range. Does not communicate well on some points (either wrong tech level or just confusing). Has technical issues such as poor speech quality on video. Developing if 2/3 of these are true                                                                                                      | Video has good quality and is in within 3-5 min range. Communicates the major results at an accessible level for non-technical people.                                                                                                                        | Video has high production quality and is in within 3-5 min range. Communicates the major results clearly and gives a sense of the methods at an accessible level for non-technical people. Has a wow factor (e.g. funny or original approach)                                                                                                                     |
| Quality modifier (can result in extra credit beyond 18 points for this assignment) | Almost everything was just barely proficient. Lots of minimal effort work. Do not double punish; if they already lost points on the rubric do not invoke this: -0.5 points                                                                                                                        |                                                                                                                                                                                                                                                                                                                                     |                                                                                                                                                                      |                                                                                                                                                                                                                                                                                                                       More than 5 excellents on this rubric: +0.5 points                                                                                   |

# COGS 108 - Final Project (change this to your project's title)

## Permissions

Place an `X` in the appropriate bracket below to specify if you would like your group's project to be made available to the public, including the names of the authors.

* [  ] YES - make available
* [  ] NO - keep private

## Link to video
https://link.to.your.publicly.viewable.video

## Abstract

Please write one to four paragraphs that describe a very brief overview of why you did this, how you did, and the major findings and conclusions.

## Authors

Instructions: REPLACE the contents of this cell with your team list and their contributions. Note that this will change over the course of the checkpoints

This is a modified [CRediT taxonomy of contributions](https://credit.niso.org). For each group member please list how they contributed to this project using these terms:
> Analysis, Background research, Conceptualization, Data curation, Experimental investigation, Methodology, Project administration, Software, Visualization, Writing – original draft, Writing – review & editing

Example team list and credits:
- Alice Anderson: Conceptualization, Data curation, Methodology, Writing - original draft
- Bob Barker:  Analysis, Software, Visualization
- Charlie Chang: Project administration, Software, Writing - review & editing
- Dani Delgado: Analysis, Background research, Visualization, Writing - original draft

## Research Question

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback


## Background and Prior Work

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback


## Hypothesis


Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback


## Data

### Data overview

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback.



### Dataset #1 (use name instead of number here)

**NOTE** The final report should not be overly long.  If this wrangling step is huge and complicated you should remove these sections from the final report and replace them with a statement like "To see the code responsible for wrangling the data see `01-DataCheckpoint.ipynb`. From here on we will simply load the processed data for further analysis"

In [ ]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION 

### Dataset #2 (if you have more than one, use name instead of number here)

In [ ]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION 

## Results

### Exploratory Data Analysis

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your EDA feedback

**NOTE** The final report should not be overly long.  If this EDA step is huge and complicated  and you went down a lot of dead ends, then you should do something about that.  Some options... 

1) remove any EDA you later decided was unecessary side quest.
2) if the code to EDA is very long (in either lines or in time to compute) make sure you write out images of the EDA into `results/` and reload them as images here instead of recalculating it. And make a statement here like "To see the code responsible for EDA analyses see `02-EDACheckpoint.ipynb`."

### First Analysis You Did - Give it a better title

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.  Every project will be different here... but some examples of analyses are: inferential statistics where you calculate p-values (like a t-test or a Monte Carlo simulation), regression models, or a machine learning model

**NOTE** The final report should not be overly long.  If this analysis step is huge and complicated  and you went down a lot of dead ends, then you should do something about that.  Some options... 

1) remove any analysis you later decided was unecessary side quest before checking in for the final hand in
2) if the code to do an analysis is very long (in either lines or in time to compute) put it in a different file.  In that file make write out results of analysis into `results/` and reload the results here instead of recalculating it. And make a statement here like "To see the code responsible for analyses see `path/to/some-file-name.ipynb_or.py`."

In [ ]:
## YOUR CODE HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION

### Second Analysis You Did - Give it a better title

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.  Every project will be different here... but some examples of analyses are: inferential statistics where you calculate p-values (like a t-test or a Monte Carlo simulation), regression models, or a machine learning model

In [ ]:
## YOUR CODE HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION

### Classification

We constructed an `access_disruption_score` from five indicators: delayed medical care, not having a usual place of care, at least one ER visit, more than one year since last doctor visit, and poor/fair self-rated health. The score ranges from 0 to 5, where higher values indicate more healthcare access disruption. First, we use Logistic Regression to build classification model, but the unbalanced class lead to low accuracy, according to the confusion matrix and classification report.


In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay
from sklearn.model_selection import train_test_split
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.pipeline import Pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC

In [6]:
clean_nhis = pd.read_csv('data/02-processed/nhis_2024_sample_adult_cleaned.csv')
df_ML=clean_nhis.copy()

df_ML['access_disruption_score'] = (
    (df_ML['delayed_medical_care'] == 1).astype(int)
    + (df_ML['usual_place_care'] != 'Has usual place of care').astype(int)
    + (df_ML['any_er_visit'] >= 1).astype(int)
    + (df_ML['doctor_gap_yr'] > 1).astype(int)
    + (df_ML['general_health'] <= 2).astype(int)
)
df_ML=df_ML.drop(columns=['government_rent_assistance'])
df_ML=df_ML.rename(columns={'AGEP_A':'age','POVRATTC_A':'poverty_ratio'})

In [7]:
df_ML['government_rent_assistance_cat']=clean_nhis['government_rent_assistance'].map({-1:'not applicable',0:'No',1:'yes'})


df_ML['health_insurance'] = clean_nhis['health_insurance'].map({
    'Uninsured': 0,
    'Insured': 1
})

X=df_ML.drop(columns=['HHX', 'WTFA_A', 'PSTRAT', 'PPSU','usual_place_care','general_health','delayed_medical_care', 'any_er_visit', 'doctor_gap_yr','access_disruption_score'])
y=df_ML['access_disruption_score']
w=df_ML["WTFA_A"]


X_train, X_test, y_train, y_test = train_test_split(
    X, y, 
    test_size=0.2,
    stratify=y,
    random_state=42
)
num_cols = X.select_dtypes(include=["int64", "float64"]).columns
cat_cols = X.select_dtypes(include=["object", "category"]).columns

In [8]:

numeric_transformer = Pipeline([
    ('scaler', StandardScaler())
])

categorical_transformer = Pipeline([
    ('onehot', OneHotEncoder(handle_unknown='ignore'))
])
preprocessor = ColumnTransformer([
    ('num', numeric_transformer, num_cols),
    ('cat', categorical_transformer, cat_cols)
])
model = Pipeline([
    ('prep', preprocessor),
    ('svm',LogisticRegression(
    max_iter=5000))
])


In [ ]:
model.fit(
    X_train,
    y_train
    )
y_pred=model.predict(X_test)

In [ ]:

class_names = np.unique(np.concatenate((y_test, y_pred)))

conf_mat = confusion_matrix(y_test, y_pred)

ConfusionMatrixDisplay(
    conf_mat,
    display_labels=class_names
).plot()

plt.xticks(rotation=45)
plt.title("Confusion Matrix")
plt.show()

from sklearn.metrics import classification_report
print(classification_report(y_test, y_pred))

### Count Model: Healthcare Access Disruption Score

Because the binary delayed-care outcome is highly imbalanced, we also modeled a broader count-style outcome instead of treating the task as a pure classification problem. 

We first considered a negative binomial model, but the fitted dispersion parameter was effectively zero, meaning the score was not meaningfully overdispersed. For this reason, the more stable pilot model reported below is a Poisson regression with robust standard errors. This keeps the model interpretable while avoiding the class-imbalance problem from the binary logistic regression.


In [9]:
from pathlib import Path
import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from IPython.display import display



analysis_data = clean_nhis.copy()
# Construct disruption score in the same way as classification analysis
analysis_data['access_disruption_score'] = (
    (analysis_data['delayed_medical_care'] == 1).astype(int)
    + (analysis_data['usual_place_care'] != 'Has usual place of care').astype(int)
    + (analysis_data['any_er_visit'] >= 1).astype(int)
    + (analysis_data['doctor_gap_yr'] > 1).astype(int)
    + (analysis_data['general_health'] <= 2).astype(int)
)

score_distribution = analysis_data['access_disruption_score'].value_counts().sort_index().to_frame('n')
score_distribution['percent'] = score_distribution['n'] / score_distribution['n'].sum() * 100

distribution_summary = pd.DataFrame({
    'mean': [analysis_data['access_disruption_score'].mean()],
    'variance': [analysis_data['access_disruption_score'].var()],
    'min': [analysis_data['access_disruption_score'].min()],
    'max': [analysis_data['access_disruption_score'].max()]
})

display(score_distribution.round(2))
display(distribution_summary.round(3))


,n,percent
access_disruption_score,,
0,15009,51.90
1,9233,31.93
2,3846,13.30
3,738,2.55
4,88,0.30
5,3,0.01


,mean,variance,min,max
0,0.675,0.677,0,5


In [10]:
full_formula = '''access_disruption_score ~ transportation_barrier + housing_cost_trouble
+ C(housing_tenure, Treatment(reference="Owned or being bought"))
+ AGEP_A + POVRATTC_A
+ C(sex) + C(race_ethnicity) + C(education) + C(health_insurance)'''

full_model = smf.glm(
    formula=full_formula,
    data=analysis_data,
    family=sm.families.Poisson()
).fit(cov_type='HC1')

renters = analysis_data[analysis_data['housing_tenure'] == 'Rented'].copy()
renter_formula = '''access_disruption_score ~ transportation_barrier + housing_cost_trouble
+ government_rent_assistance + AGEP_A + POVRATTC_A
+ C(sex) + C(race_ethnicity) + C(education) + C(health_insurance)'''

renter_model = smf.glm(
    formula=renter_formula,
    data=renters,
    family=sm.families.Poisson()
).fit(cov_type='HC1')

def make_irr_table(result, model_name):
    conf = result.conf_int()
    table = pd.DataFrame({
        'model': model_name,
        'term': result.params.index,
        'coef': result.params.values,
        'IRR': np.exp(result.params.values),
        'CI_lower': np.exp(conf[0].values),
        'CI_upper': np.exp(conf[1].values),
        'p_value': result.pvalues.values
    })
    return table

full_results = make_irr_table(full_model, 'Full sample Poisson')
renter_results = make_irr_table(renter_model, 'Renter-only Poisson')
poisson_results = pd.concat([full_results, renter_results], ignore_index=True)

key_terms = [
    'transportation_barrier',
    'housing_cost_trouble',
    'C(housing_tenure, Treatment(reference="Owned or being bought"))[T.Rented]',
    'C(housing_tenure, Treatment(reference="Owned or being bought"))[T.Other arrangement]',
    'government_rent_assistance'
]
key_results = poisson_results[poisson_results['term'].isin(key_terms)].copy()

Path('results').mkdir(exist_ok=True)
poisson_results.to_csv('results/access_disruption_poisson_results.csv', index=False)
key_results.to_csv('results/access_disruption_poisson_key_results.csv', index=False)

display(key_results[['model', 'term', 'IRR', 'CI_lower', 'CI_upper', 'p_value']].round(4))


,model,term,IRR,CI_lower,CI_upper,p_value
1,Full sample Poisson,"C(housing_tenure, Treatment(reference=""Owned o...",1.2435,1.1493,1.3455,0.0000
2,Full sample Poisson,"C(housing_tenure, Treatment(reference=""Owned o...",1.2513,1.2135,1.2903,0.0000
19,Full sample Poisson,transportation_barrier,1.3417,1.2839,1.4022,0.0000
20,Full sample Poisson,housing_cost_trouble,1.3879,1.3318,1.4465,0.0000
40,Renter-only Poisson,transportation_barrier,1.3605,1.2837,1.4420,0.0000
41,Renter-only Poisson,housing_cost_trouble,1.2857,1.2169,1.3583,0.0000
42,Renter-only Poisson,government_rent_assistance,1.0586,0.9986,1.1223,0.0558


The pilot count model gives a more detailed result and trend compared to exploratory analysis. After adjusting for controls (age, poverty ratio, sex, race/ethnicity, education, and insurance), respondents with a transportation barrier had about **34% higher expected healthcare access disruption** than respondents without that barrier (IRR = 1.34, 95% CI [1.28, 1.40], p < 0.001). Respondents who had trouble paying housing costs had about **39% higher expected disruption** (IRR = 1.39, 95% CI [1.33, 1.45], p < 0.001). Compared with homeowners, renters had about **25% higher expected disruption** (IRR = 1.25, 95% CI [1.21, 1.29], p < 0.001).

As a renter-only reliability check, government rent assistance was associated with a small increase in expected disruption, but this estimate was only borderline significant (IRR = 1.06, 95% CI [1.00, 1.12], p = 0.056). This suggests rent assistance may identify a more vulnerable subgroup of renters, but it is not as strong or stable a predictor as transportation barriers and housing cost trouble. Overall, this model appears more appropriate than a classification model because it preserves the direction and magnitude of the housing and transportation effects without being dominated by the majority class.


### Examine the dataset with with most dominant ethnicity (White) removed

In [15]:

not_white = analysis_data[analysis_data['race_ethnicity'] != 'White non-Hispanic'].copy()

not_white_model = smf.glm(
    formula=full_formula,
    data=not_white,
    family=sm.families.Poisson()
).fit(cov_type='HC1')

nonwhite_renters=not_white[not_white['housing_tenure'] == 'Rented'].copy()
nonwhite_renter_model = smf.glm(
    formula=renter_formula,
    data=nonwhite_renters,
    family=sm.families.Poisson()
).fit(cov_type='HC1')



not_white_results = make_irr_table(not_white_model, 'Non-white subsample')
nonwhite_renter_results = make_irr_table(nonwhite_renter_model, 'Nonwhite_renter-only Poisson')
poisson_results   = pd.concat([not_white_results,nonwhite_renter_results], ignore_index=True)


key_results = poisson_results[poisson_results['term'].isin(key_terms)].copy()

Path('results').mkdir(exist_ok=True)
poisson_results.to_csv('results/access_')
display(key_results[['model', 'term', 'IRR', 'CI_lower', 'CI_upper', 'p_value']].round(4))

,model,term,IRR,CI_lower,CI_upper,p_value
1,Non-white subsample,"C(housing_tenure, Treatment(reference=""Owned o...",1.1680,1.0045,1.3582,0.0436
2,Non-white subsample,"C(housing_tenure, Treatment(reference=""Owned o...",1.1752,1.1159,1.2376,0.0000
18,Non-white subsample,transportation_barrier,1.2956,1.2082,1.3892,0.0000
19,Non-white subsample,housing_cost_trouble,1.3155,1.2337,1.4027,0.0000
38,Nonwhite_renter-only Poisson,transportation_barrier,1.3029,1.1959,1.4194,0.0000
39,Nonwhite_renter-only Poisson,housing_cost_trouble,1.2072,1.1159,1.3061,0.0000
40,Nonwhite_renter-only Poisson,government_rent_assistance,1.0919,1.0031,1.1886,0.0422


### ETC AD NASEUM

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.

In [ ]:
## YOUR CODE HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION

## Ethics

Instructions: REPLACE the contents of this cell with your work, including any updates to recover points lost in your proposal feedback


## Discusison and Conclusion

Wrap it all up here.  Somewhere between 3 and 10 paragraphs roughly. 
1. Summarize the important results you made
2. This is a good time to refer back to your Background section and review how this work extended the previous stuff.
3. Try to bring all the different results together into a bigger picture... what have we learned?
4. Often what we learned is confusing.  Discuss if there are contradictory results how that might have happened
5. Discuss the limitations of your work. e.g, is the dataset a biased sample? would more data make you more confident? would longitudinal instead of cross-sectional data helpful? is there evidence that the phenomena you're investigating may change over time? etc
6. If you would like, you can also say what the next steps in this investigation might be if the work was to continue outside this class
